### DISCUSSION TITLE
Configure severity of analyzer to report only specific location

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/roslyn#74851] [Q&A] [answered]
This code shows a CS8524 and CS8509 warning, since the switch is not exhaustive and also requires a default:

CS8524: The switch expression does not handle the default case or unnamed enums.
CS8509: The switch expression does not handle all possible values

```cs
  public static bool IsRequest(this CommandField command)
  {
      // Shows squiggles CS8524 that the switch is not exhaustive.
      return command switch
      {
          CommandField.C_STORE_RQ   => true,
          CommandField.C_STORE_RSP  => false,
      };
  }
```

The problem is that I don't want a default, since everytime a new enum value is added it is not found and and to write tests if every enum value is treated correctly is too much effort since we have thousands of different enums. I really want to have all cases treated and unnamed cases should throw an exception. So there is already a distinction with CS8524 and CS8509.

I can configure both with th .editorconfig but it is not possible to change the analyzer behavior in code.

There is `System.Diagnostics.CodeAnalysis.SuppressMessageAttribute` but what I'm missing is to set some severity, or better to define the severity, for example as a `System.Diagnostics.CodeAnalysis.SeverityAttribute`.

Do I miss something?

**Comment by 333fred** [comment, marked as answer]
Editorconfig and/or #pragmas are the solution here.

**Reply by msedi** [reply]
@333fred: Understood, but if I'm correct with `#pragma` I cannot change the severity, right? 
Which would mean I must set `CS8524` as error in editorconfig and mark all existing switch cases with `#pragma disable` which would then prevent all warnings.

**Reply by 333fred** [reply]
That's correct.
